pkg/ipam/storage/sqlite/sqlite.go

```diff
@@ -51,6 +58,197 @@ func New(datastore string) (*SQLiteIPAMStorage, error) {
 	return sqlis, nil
 }
 
+// StartGarbageCollector -
+// StartGarbageCollector starts a one time garbage collector to remove
+// leftover data created based on the old model. Also, periodically runs
+// a garbage collector to clean up outdated records with valid updatedAt
+// values (where threshold determines what is considered outdated).
+func (sqlis *SQLiteIPAMStorage) StartGarbageCollector(ctx context.Context, interval time.Duration, threshold time.Duration) {
+	var once sync.Once
+	go func() {
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+
+		/*
+			initialDelay := 1 * time.Minute
+			timer := time.NewTimer(initialDelay)
+			select {
+			case <-timer.C:
+			case <-ctx.Done():
+				timer.Stop()
+				select {
+				case <-timer.C:
+				default:
+				}
+				return
+			} */
+
+		// Run the periodic task
+		for {
+			select {
+			case <-ctx.Done():
+				return // Exit the goroutine when the stop signal is received
+			case <-ticker.C:
+				once.Do(func() {
+					if err := sqlis.oneTimeGarbageCollector(ctx); err != nil {
+						log.Logger.Info("One time garbage collector returned error", "err", err)
+					}
+				})
+				if err := sqlis.garbageCollector(ctx, threshold); err != nil {
+					log.Logger.Info("Garbage collector returned error", "err", err)
+				}
+			}
+		}
+	}()
+}
+
+// Fetch and delete -src/-dst records whose updatedAt timestamp is considered
+// expired according to the threshold.
+func (sqlis *SQLiteIPAMStorage) garbageCollector(ctx context.Context, threshold time.Duration) error {
```

Comment:
both `oneTimeGarbageCollector` and `garbageCollector` might be the same function except the sql  query